util/formula_templater/formula_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrintFormula(t *testing.T) {
+	product := "consul"
+	version := "1.9.4"
+	config := "./config.hcl"
+	buf := new(bytes.Buffer)
+	expect := `class Consul < Formula
+  desc "Consul"
+  homepage "https://www.consul.io"
+  version "1.9.4"
+
+  if OS.mac? && Hardware::CPU.intel?
+    url "https://releases.hashicorp.com/consul/1.9.4/consul_1.9.4_darwin_amd64.zip"
+    sha256 "c168240d52f67c71b30ef51b3594673cad77d0dbbf38c412b2ee30b39ef30843"
+  end
+
+  if OS.linux? && Hardware::CPU.intel?
+    url "https://releases.hashicorp.com/consul/1.9.4/consul_1.9.4_linux_amd64.zip"
+    sha256 "da3919197ef33c4205bb7df3cc5992ccaae01d46753a72fe029778d7f52fb610"
+  end
+
+  if OS.linux? && Hardware::CPU.arm? && Hardware::CPU.is_64_bit?
+    url "https://releases.hashicorp.com/consul/1.9.4/consul_1.9.4_linux_arm64.zip"
+    sha256 "012c552aff502f907416c9a119d2dfed88b92e981f9b160eb4fe292676afdaeb"
+  end
+
+  bottle :unneeded
+
+  conflicts_with "consul"
+
+  def install
+    bin.install "consul"
+  end
+
+  plist_options manual: "consul agent -dev -bind 127.0.0.1"
+
+  def plist; <<~EOS
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>KeepAlive</key>
+    <dict>
+        <key>SuccessfulExit</key>
+        <false/>
+    </dict>
+    <key>Label</key>
+    <string>#{plist_name}</string>
+    <key>ProgramArguments</key>
+    <array>
+        <string>#{opt_bin}/consul</string>
+        <string>agent</string>
+        <string>-dev</string>
+        <string>-bind</string>
+        <string>127.0.0.1</string>
+    </array>
+    <key>RunAtLoad</key>
+    <true/>
+    <key>WorkingDirectory</key>
+    <string>#{var}</string>
+    <key>StandardErrorPath</key>
+    <string>#{var}/log/consul.log</string>
+    <key>StandardOutPath</key>
+    <string>#{var}/log/consul.log</string>
+</dict>
+</plist>
+
+EOS
+  end
+
+  test do
+    system "#{bin}/consul --version"
+  end
+end
+`
+
+	err := printFormula(product, version, config, buf)
+	require.Nil(t, err)
+	assert.Equal(t, expect, buf.String())
+}
```

Comment:
Nice test, how often might this need to be updated? Just thinking the SHAs might be a pain to update. One way around this could be to use an `-update` test flag to update the expected data (in a separate file rather than inline). I used this in packagespec, adding [the flag here](https://github.com/hashicorp/packagespec/blob/main/cmd/packagespec/main_test.go#L17) then using a [helper](https://github.com/hashicorp/packagespec/blob/main/cmd/packagespec/lockcommand_test.go#L88) to do assertions in the test that [updates the test data](https://github.com/hashicorp/packagespec/blob/main/cmd/packagespec/main_test.go#L71-L74) if the flag is set, before doing the comparison. (That helper diffs an entire filesystem hierarchy, so something simpler would work here.) Just a thought if this becomes difficult to manage in the future.

Reply:
The original intent was to have this never (or rarely) change as this should be static for a specific version, but I could definitely see value in periodically updating the version we're testing against.